### ISSUE TITLE
What's the minimum rustc version supported?

### ISSUE THREAD

**Issue by behnam** [killercup/assert_cli#78]
With `rust-1.17.0` or `rust-1.18.0`, I'm get this error on build time:

```rust
error: use of unstable library feature 'command_envs' (see issue #38526)
   --> /Users/behnam/.cargo/registry/src/github.com-1ecc6299db9ec823/assert_cli-0.5.4/src/assert.rs:303:14
    |
303 |             .envs(self.env.clone().compile())
    |              ^^^^
```

It's fine on stable channel, which is at `rust-1.21.0` right now. So, the minimum version supported in larger that `1.18.0`, less than or equal to `1.21.0`.

Looking at the `.travis.yml` file here, I noticed there's no minimum rustc version being maintained.

Could you please indicate what's the minimum rustc version supported by `assert_cli`?

FYI, if supporting versions a few months old is not a goal here, I will have to to gate my cli test behind a feature, which would make my CI script more complicated.

So, any chance we can have some versions older than stable also supported?

----

- `assert_cli` version: v0.5.4
- Rust version: 1.17.0
- OS and version: Darwin Marv.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64

**Comment by epage** [comment]
Looks like that feature was stablized in 1.19
https://github.com/rust-lang/rust/blob/master/RELEASES.md#stabilized-apis-2

What is your use case for holding back your rustc version?

**Comment by behnam** [comment]
Right. I looked into the code and noticed that `assert_cli` has a bunch of features from 1.19.0.

I'm tracking Firefox's minimum rustc version for UNIC, as also do many Servo projects. Here's the status for Firefox: https://wiki.mozilla.org/Oxidation

Basically, it's currently 1.17.0, but in a few days will become 1.19.0.

In many projects, we add a specific rust version to `.travis.yml`, which will make sure new code is always compatible. Would love to see that happen in `assert_cli`, as it makes the project more reliable.

**Comment by epage** [comment]
Thanks.  I hear of discussions of libraries supporting old stable versions but haven't heard too much the use case of why their users need them.

I think other crates are doing a `stable-2`.  Something for us to consider.
(and yes, if we do it, we would have the CI enforce it).

**Comment by killercup** [comment]
Looking into the future a bit, I don't think we'll need any of the features
stabilized in 1.23. Not sure about our dependencies, though.

I'm fine with adding a 1.19 build to Travis.

Ed Page <notifications@github.com> schrieb am Di. 7. Nov. 2017 um 06:27:
